Highlands dimension

John-Candlebury opened the pull request:
#### Summary
Content "Highlands Dimension"

#### Purpose of change
Create a new dimension that is visible through gameplay mishaps. In this case you can get here through being outside in a portal storm with accumulated IRE. You'll return so long as you arent portalproof in a random amount of time no longer than 2 days.

#### Describe the solution
A predominantly flat landscape of tall grassland  and dirt dotted with ravines and small mounds.  Its pretty windy and it rains almost all the time, exposure is a risk. Theres no animals or foes within it (the shadow can probably spawn here actually).

It has ancient ruins on it too.

Heres a pretty typical segment of the dimension:

<img width="1855" height="1011" alt="image" src="https://github.com/user-attachments/assets/8091a5ba-7e6b-47d9-95dd-a824f5f48463" />

Here's a less typical one
<details><img width="1519" height="1007" alt="image" src="https://github.com/user-attachments/assets/8a1fef36-98ac-43ac-9291-bcaac836057d" />
</details>

#### Testing
Ran the EOCs manually to make sure they worked.

#### Additional context
It seems you currently cannot place specials with cities disabled, I wanted to include some ruins with this, but its not currently possible.

John-Candlebury committed:
Highlands dimension

PatrikLundell commented:
What's the point of this? It looks like you're just thrown into a prison of boredom for a random amount of time (with an exposure risk if it happens during summer, as you were expecting to battle heat, not cold). I assume there's no return to that particular place, so any work you do to prepare a shelter is wasted time.

Standing-Storm commented:
I really like this idea but I think it could have a `"one_in_chance"` field like Yrax attention to make it rarer. As it is, with the portal storm EoC checking every 35 seconds, the odds of you being sent on a dimensional jaunt during a storm are pretty high.

John-Candlebury committed:
Allow placement of specials, add roman roads

John-Candlebury committed:
Disable highland specials on other regions

John-Candlebury committed:
Only reach the highlands through shallow water

John-Candlebury committed:
Add rare temples

OtherBiscuit commented:
I like the idea of this mini-dimension being more of an emergency escape. Like-

You're bogged down and know you can't make it through to the exit, but there _is_ a portal to a potentially hostile world. Between certain death and uncertain death, you take your chances and end up in an unfamiliar land.
If you can survive the elements for a while, you can get back on your feet and lick your wounds while you explore this strange environment. If not, at least you bought yourself a few more hours.

The emptiness is a great aesthetic though. Empty hills for miles and miles, only a cobblestone path to nowhere to guide you. I feel like sparse buildings might ruin it (pun intended, hehe ruins) but this could be a great chance for some locations that are just environmental. Finding a bog, or a forest, or something in a valley. Silent woods and bogs are creepy, where are the bugs and birds?

Perhaps you find a puddle with more... depth, near a lake of strange liquid. 
`You sink through and slowly feel as though the liquid is getting thicker, your mind slows as the puddle gives more resistance than just physical. You slowly come to, hazy, in the same spot you entered the first portal.`

I don't know, spitballing ideas.

John-Candlebury committed:
Create an achievement for the highlands

John-Candlebury committed:
Solve several issues

Solve possible load order error

Update highlands.json

Test

Fix mutable placement error

Update terrain-fences-gates.json

Fixes
